Bio/PDB/StructureAlignment.py

```diff
@@ -7,28 +7,57 @@
 
 """Map residues of two structures to each other based on a FASTA alignment."""
 
+from typing import Optional
+
+from Bio.Align import Alignment, MultipleSeqAlignment, PairwiseAligner
 from Bio.Data import PDBData
 from Bio.PDB import Selection
+from Bio.PDB.Model import Model
 from Bio.PDB.Polypeptide import is_aa
+from Bio.PDB.Residue import Residue
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
 
 
 class StructureAlignment:
     """Class to align two structures based on an alignment of their sequences."""
 
-    def __init__(self, fasta_align, m1, m2, si=0, sj=1):
+    def __init__(
+        self,
+        fasta_align: Optional[MultipleSeqAlignment | Alignment] = None,
+        m1: Model | None = None,
```

Comment:
I'd add a deprecation warning stating that `fasta_align` will become an optional argument and that the function signature will change accordingly in a later release. This can give people some warning to adjust their code in the wild.